### ISSUE TITLE
Launching pcluster with spot type, awsbatch and pre-configured AMI takes very long time.

### ISSUE THREAD

**Issue by rangsimanketkaew** [aws/aws-parallelcluster#1642]
**Environment:**
 - AWS ParallelCluster v. 2.5.1
 - Config file I am using is below

```
[global]
cluster_template = default
update_check = true
sanity_check = true

[aws]
aws_region_name = us-east-1

[cluster default]
key_name = bla_bla_bla
custom_ami = ami-XXXXXXXXXXXXX
master_instance_type = g3.8xlarge
compute_instance_type = g3.8xlarge
master_root_volume_size = 30
compute_root_volume_size = 30
initial_queue_size = 0
max_queue_size = 8
max_vcpus = 128
maintain_initial_size = false
scheduler = awsbatch
cluster_type = spot
placement_group = DYNAMIC
placement = cluster
ebs_settings = shared
vpc_settings = public
spot_bid_percentage = 85
spot_price = 0.7

[ebs shared]
shared_dir = my_shared
volume_type = st1
volume_size = 500

[vpc public]
vpc_id = vpc-XXXXXXXXX
master_subnet_id = subnet-XXXXXXX

[aliases]
ssh = ssh {CFN_USER}@{MASTER_IP} {ARGS}

[scailing custom]
scaledown_idletime = 3
```

**Problem description and how to reproduce:**
I had launched a parallel cluster using pcluster 2.5.1 with modified configurations including using my pre-installed AMI (Ubuntu 18.04), spot cluster type, and awsbatch as a job scheduler.

The pcluster took very long time to create the cluster, especially it was stuck at the step called **AWSBatchStack**. The following is stdout messages:

```
$ pcluster create cluster -c config
Beginning cluster creation for cluster: cluster
Creating stack named: parallelcluster-cluster
Status: ComputeServerLaunchTemplate - CREATE_COMPLETE
Status: AWSBatchStack - CREATE_COMPLETE
```

I am wondering that what did I do wrong? and how to solve this problem to get the cluster successfully created? I am also not sure that if custom_ami and awsbatch caused this problem.

Thank you very much.

**Comment by rexcsn** [comment]
Hi, 

I created a cluster successfully with the config file you provided, without the `custom_ami` option. I suggest to double check the custom AMI that you are using, please follow this guide to create the custom AMI: https://docs.aws.amazon.com/parallelcluster/latest/ug/tutorials_02_ami_customization.html

A minor comment, some of the parameters you are providing in the config are not applicable to `awsbatch` and you can remove them from the config, see https://docs.aws.amazon.com/parallelcluster/latest/ug/cluster-definition.html for more detail.

These parameters will be ignored automatically and should not affect cluster creation, here is the equivalent config with non-awsbatch parameters removed:
```
[cluster default]
key_name = xxx
custom_ami = ami-XXXXXXXXXXXXX
master_instance_type = g3.8xlarge
compute_instance_type = g3.8xlarge
master_root_volume_size = 30
compute_root_volume_size = 30
min_vcpus = 0
max_vcpus = 128
scheduler = awsbatch
cluster_type = spot
placement_group = DYNAMIC
placement = cluster
ebs_settings = shared
vpc_settings = default
spot_bid_percentage = 85
```

**Comment by rangsimanketkaew** [comment]
Hi rexcsn, 

Thank you a lot for your help. Now I am able to launch a spot parallel cluster with SLURM schedule and the cluster works well.

Unfortunately, I have another problem about job submission in parallel run. I submitted the job using sbatch command requesting 2 compute nodes for the job. The cluster with auto scaling tried to build new compute nodes to run that job, but the compute node had been started and stopped over and over. Then the job was hanged in the job queue, and eventually I needed to kill the job myself.

Have you ever experienced this problem? Is so, what should I do? If not, should I create a new issue for it?

Nutt

**Comment by rexcsn** [comment]
Hi,

After reading your initial ticket again I noticed that you are using a pre-installed Ubuntu 18.04 AMI. To clarify, `awsbatch` only works with `alinux` at the moment, so if you wanted to use `awsbatch` the Ubuntu 18.04 AMI definitely cannot be used.

When you created the slurm cluster I assume you are using the same Ubuntu 18.04 AMI, please double check to make sure you are specifying the `base_os` parameter as `ubuntu1804`. If not specified, `base_os` defaults to `alinux`, and if you are using an Ubuntu 18.04 AMI this inconsistency could break the cluster.

**Comment by rangsimanketkaew** [comment]
rexcsn,

Now the cluster with that SLURM scheduler could be created successfully. I also found a type in mu config file: it must have been **[scaling custom]**. Not **[scailing custom]**.

Thanks again for your help. :)

Nutt

**Comment by rexcsn** [comment]
Great! I am closing this issue, feel free to re-open is there is any question.
